### PR TITLE
drivers: hwinfo: implemented hardware info support for Smartbond

### PR DESCRIPTION
**Pull request by kasjer**
Only reset cause is supported as there is no common unique id present on those chips.
Unique ID can be put in OTP but there is no single specification for this.

### CODE REVIEW

**Comment by kasjer** [comment]
It's hard to make anyone here to review things, let's prolong for another 60 days.

File: drivers/hwinfo/hwinfo_smartbond.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+int z_impl_hwinfo_get_reset_cause(uint32_t *cause)
+{
+	int ret = 0;
+	uint32_t reason = CRG_TOP->RESET_STAT_REG;
+	uint32_t flags = 0;
+
+	/*
+	 * When POR is detected some other bits are also set that
+	 * are not really valid so they are skipped.
```
Comment (alexanderwachter) [inline drivers/hwinfo/hwinfo_smartbond.c:19]: What dies that mean "are not really valid"?
Is there a chip errata or something like that?

File: drivers/hwinfo/Kconfig [hunk 2]
```diff
@@ -150,6 +150,13 @@ config HWINFO_SAM0
 	help
 	  Enable Atmel SAM0 hwinfo driver.
 
+config HWINFO_SMARTBOND
+	bool "Smartbond device ID"
```
Comment (alexanderwachter) [inline drivers/hwinfo/Kconfig:154]:
```suggestion
	bool "Smartbond reset cause"
```

File: drivers/hwinfo/Kconfig [hunk 3]
```diff
@@ -150,6 +150,13 @@ config HWINFO_SAM0
...
+	default y
+	depends on SOC_FAMILY_SMARTBOND
+	help
+	  Enable Smartbond hwinfo driver.
```
Comment (alexanderwachter) [inline drivers/hwinfo/Kconfig:158]: Please add a note that only reset cause is supported

File: drivers/hwinfo/hwinfo_smartbond.c [hunk 4]
```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2023 Jerzy Kasenberg.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/hwinfo.h>
+#include <DA1469xAB.h>
```
Comment (alexanderwachter) [inline drivers/hwinfo/hwinfo_smartbond.c:9]: This looks very particular too me. Usually we include soc.h

**Review by alexanderwachter — CHANGES_REQUESTED** [review]

File: drivers/hwinfo/hwinfo_smartbond.c (hunk 4 above)
Comment (kasjer) [reply drivers/hwinfo/hwinfo_smartbond.c]: changed to soc.h

File: drivers/hwinfo/Kconfig (hunk 3 above)
Comment (kasjer) [reply drivers/hwinfo/Kconfig]: fixed

File: drivers/hwinfo/Kconfig (hunk 2 above)
Comment (kasjer) [reply drivers/hwinfo/Kconfig]: suggestion applied

File: drivers/hwinfo/hwinfo_smartbond.c (hunk 1 above)
Comment (kasjer) [reply drivers/hwinfo/hwinfo_smartbond.c]: Comment reworded, should be more readable now

**Comment by alexanderwachter** [comment]
@kasjer thanks for the quick fixes. Were you able to test on HW?

**Comment by alexanderwachter** [comment]
@kasjer are there boards in the tree where you could add the hwinfo as supported features?

**Comment by kasjer** [comment]
@alexanderwachter I tested it on da1469x_dk_pro, could you give me a hint on how add supported feature (I'm very new to zephyr)

**Comment by alexanderwachter** [comment]
Just add - hwinfo here:
boards/arm/da1469x_dk_pro/da1469x_dk_pro.yaml
And here:

**Review by alexanderwachter — APPROVED** [review]
Thanks @kasjer sorry for the delay!

**Comment by kasjer** [comment]
@alexanderwachter I added hwinfo feature to both boards. I had to rebase branch to include da14695_dk_usb board though so force-pushed changes may look unrelated

**Comment by kasjer** [comment]
> Thanks @kasjer sorry for the delay!

No problem I added this because I really needed to know reset cause in work on other PR so there was no rush, thanks for review and for your help with polishing things.

**Review by andrzej-kaczmarek — APPROVED** [review]

**Pull request merged**